### PR TITLE
Full basic test coverage (#26)

### PR DESCRIPTION
**Pull request by dergluck**
Hi,

I have added an additional numerics structure that contains plain and pointers values. Also initialization code added to the test function. Test snapshots eyeballed and updated.

Hope this works as requested.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1,5 +1,6 @@
 github.com/bradleyjkemp/cupaloy/v2 v2.4.0 h1:ck3EgwgVgIBuuOvmhK3oHwPP5KH//xcMTWcXERPQ9hY=
 github.com/bradleyjkemp/cupaloy/v2 v2.4.0/go.mod h1:TD5UU0rdYTbu/TtuwFuWrtiRARuN7mtRipvs/bsShSE=
```
Comment (bradleyjkemp) [inline go.sum]: I think if you run `go mod tidy` this will clean up the older cupaloy version

File: .snapshots/memviz_test-TestMap [hunk 2]
```diff
@@ -1,11 +1,11 @@
 digraph structs {
   node [shape=Mrecord];
-  3:<4key0> -> 1:name;
-  3 [label="<name> structMap |{<f0> id | \"leaf\"} |{<f1> links | {{<4key0> *memviz_test.structMap| <4value0> true}}}  "];
-  5 [label="<name> structMap |{<f0> id | \"leaf2\"} |{<f1> links | map[*memviz_test.structMap]bool\{\}}  "];
+  3 [label="<name> structMap |{<f0> id | \"leaf2\"} |{<f1> links | map[*memviz_test.structMap]bool\{\}}  "];
```
Comment (bradleyjkemp) [inline .snapshots/memviz_test-TestMap]: This change just happens because maps don't have deterministic output (yet). Can you revert the changes to this file?

Comment (dergluck) [reply .snapshots/memviz_test-TestMap]: Hi, done it.

**Comment by dergluck** [comment]
Hi, the changes are done as suggested
